app/assets/javascripts/osom-tables.js

```diff
@@ -117,6 +132,32 @@
     }
   }
 
+  function save_checkboxes(container) {
+    var checked_ids;
+    checked_ids = (container.data('checked') || '').split(',');
+    container.find('td.mark input[type="checkbox"]').each(function() {
+      var item_id;
+      item_id = $(this).data('item-id').toString();
+      if ($.inArray(item_id, checked_ids) === -1) {
+        if (this.checked) {
+          return checked_ids.push(item_id);
+        }
+      } else if (!this.checked) {
+        return checked_ids.splice($.inArray(item_id, checked_ids), 1);
+      }
+    });
+    return container.data('checked', checked_ids.join(',').replace(/^,/, ''));
+  };
+
+  function restore_checkboxes(container) {
```

Comment:
Also, what does "restore checkboxes" mean? Is this like `load_checkbox_state_for_page` or something?